test/test_infer_util.py

```diff
@@ -253,3 +253,17 @@ def model(data):
         for name, p in init_params[0].items():
             # XXX: the result is equal if we disable fast-math-mode
             assert_allclose(p[i], init_params_i[0][name], atol=1e-6)
+
+
+def test_improper_expand():
+
+    def model():
+        population = np.array([1000., 2000., 3000.])
+        with numpyro.plate("region", 3):
+            numpyro.sample("incidence",
+                           dist.ImproperUniform(support=constraints.interval(0, population),
+                                                batch_shape=(3,),
+                                                event_shape=(3,)))
+
+    model_info = initialize_model(random.PRNGKey(0), model)
+    assert model_info.param_info.z['incidence'].shape == (3, 3)
```

Comment:
What I wanted to verify for this example is that with `event_shape=()`, this should be  (3,). Is that correct?

Reply:
Let me test it. I don't know what happens when users provide an invalid event_shape.

Reply:
Thanks, @neerajprad! I just added that test case and fixed the issue at `init_to_uniform`. Does the fix sound correct to you?

Reply:
Great, that makes sense.